data/src/main/java/org/apache/iceberg/data/orc/GenericOrcWriters.java

```diff
@@ -231,7 +231,14 @@ public void nonNullWrite(int rowId, String data, ColumnVector output) {
 
     @Override
     public void nonNullWrite(int rowId, ByteBuffer data, ColumnVector output) {
-      ((BytesColumnVector) output).setRef(rowId, data.array(), 0, data.array().length);
+      ByteBuffer dupe = data.duplicate();
+      if (dupe.hasArray()) {
+        ((BytesColumnVector) output).setRef(rowId, dupe.array(), dupe.arrayOffset(), dupe.remaining());
```

Comment:
The offset should be `dupe.arrayOffset() + dupe.position()`.

Reply:
Ok. I have updated 👍